openff/toolkit/tests/test_forcefield.py

```diff
@@ -4322,6 +4321,72 @@ def test_overwrite_bond_orders(self):
         ):
             assert bond1.bond.fractional_bond_order == bond2.bond.fractional_bond_order
 
+    def test_fractional_bond_order_ignore_existing_confs(self):
+        """Test that previously-defined bond orders in the topology are overwritten"""
+        mol = create_ethanol()
+        top = Topology.from_molecules(mol)
+
+        mod_mol = create_ethanol()
+        mod_mol.generate_conformers()
+        mod_mol._conformers[0][0][0] = (
+            mod_mol._conformers[0][0][0] + 1.0 * unit.angstrom
+        )
+        mod_mol._conformers[0][1][0] = (
+            mod_mol._conformers[0][1][0] - 1.0 * unit.angstrom
+        )
+        mod_mol._conformers[0][2][0] = (
+            mod_mol._conformers[0][2][0] + 1.0 * unit.angstrom
+        )
+
+        mod_top = Topology.from_molecules(mod_mol)
+
+        forcefield = ForceField("test_forcefields/test_forcefield.offxml", xml_ff_bo)
+
+        omm_system, omm_sys_top = forcefield.create_openmm_system(
+            top, return_topology=True
+        )
+        mod_omm_system, mod_omm_sys_top = forcefield.create_openmm_system(
+            mod_top, return_topology=True
+        )
+
+        # Check that the assigned bond parameters are identical for both systems
+        default_bond_force = [
+            f for f in omm_system.getForces() if isinstance(f, openmm.HarmonicBondForce)
+        ][0]
+        mod_bond_force = [
+            f
+            for f in mod_omm_system.getForces()
+            if isinstance(f, openmm.HarmonicBondForce)
+        ][0]
+
+        for idx in range(default_bond_force.getNumBonds()):
+            assert default_bond_force.getBondParameters(
+                idx
+            ) == mod_bond_force.getBondParameters(idx)
```

Comment:
In general non-exact values should be compared with something like `np.isclose` or `np.allclose` to avoid potential floating point issues.

```suggestion
            assert all(
                np.isclose(
                    default_value.value_in_unit(default_value.unit),
                    mod_value.value_in_unit(default_value.unit),
                )
                for default_value, mod_value in zip(
                    default_bond_force.getBondParameters(idx)[2:],
                    mod_bond_force.getBondParameters(idx)[2:],
                )
            )
```

It may be good to add equivalent utility methods (separate PR) which perform `isclose` and `allclose` on `simtk.unit.Quantity` objects.